### ISSUE TITLE
make it automatically build on appveyor

### ISSUE THREAD

**Issue by rgl** [coddec/Classic-Shell#6]
it would be nice to have this code build automatically on appveyor.com.

**Comment by coddec** [comment]
@rgl 

> it would be nice to have this code build automatically on appveyor.com.

Created one, not sure if it will work until there's some new code though I guess

**Comment by coddec** [comment]
Everything configured, get following error, tried to upgrade from VS2008 to VS2017 project locally, brings up some warnings. Instead of break anything, I'll wait & see if anyone else can do it correctly without break anything and hopefully can share the procedure with me.

https://ci.appveyor.com/project/coddec/classic-shell/branch/master
`The project file "ClassicExplorer\ClassicExplorer.vcproj" is in the ".vcproj" file format, which MSBuild no longer supports. Please convert the project by opening it in the Visual Studio IDE or running the conversion tool, or use MSBuild 3.5 or earlier to build it.`

# AppVeyor build screenshot
![Screen](https://images2.imgbox.com/06/bd/pozemZzl_o.png)

**Comment by war59312** [comment]
@coddec Could probably get assistance @ https://help.appveyor.com/discussions/problems/7200-specify-which-project-or-solution-file-to-use-because-the-folder-contains-more-than-one-project-or-solution-file

**Comment by coddec** [comment]
@war59312 Thanks for the link, the error is about the version of Visual Studio the project used, I've already resolved multiple project issue mentioned in the link you given. ;)
Cheers

**Comment by IkeKap** [comment]
@coddec what were the specific warnings Visual studio displayed during the conversion process? Also would it be possible to branch off the converted file so it would be easier for multiple people to attempt to resolve the issues as they appear?

**Comment by IkeKap** [comment]
It seems that allowing pull request #7 would open the way to solving this issue @coddec

**Comment by coddec** [comment]
@IkeKap I agree, just waiting someone who can review the changes just in case then we can merge it, hopefully that will resolve this issues too ;)

**Comment by H3rmit** [comment]
so, is anyone going to keep this project alive? I really hope so!

**Comment by XenHat** [comment]
I cannot afford appveyor membership at the moment so my help will be very limited however there is a template for the `.appveyor.yaml` file [here](https://www.appveyor.com/docs/appveyor-yml/) and some alternate instructions from the support staff concerning **multiple projects building** have been provided [in this answer](https://help.appveyor.com/discussions/questions/1544-multiple-solutions-in-a-repository#comment_37538216).

TL;DR:
```yaml
    matrix: 
      - project_name: src/Project1/XXXX.YYY.Project1.sln 
      - project_name: src/Project2/XXXX.YYY.Project2.sln 
      - project_name: src/Project3/XXXX.YYY.Project3.sln 
      - project_name: src/Project4/XXXX.YYY.Project4.sln 
      - project_name: Shared/Foldername/somesolution.sln
```
and then
```yaml
  build: 
    verbosity: normal 
    project: $(project_name)
```

**Comment by coddec** [comment]
@XenHat Thanks for the approval and information for appveyor, since this project is open source, if I remember right, we can use appreyor for free ;) so.. no need to pay for anything. Don't worry 😄

**Comment by coddec** [comment]
Ummmmm, new error?
![screenshot-2018-2-2 classic shell 4 3 1 1 - appveyor](https://user-images.githubusercontent.com/12678338/35716417-f5bf6076-082c-11e8-9aca-c066213baa81.png)

**Comment by XenHat** [comment]
I wonder if appveyor cleaned the previous artifacts mid-build.. 🤔

**Comment by ge0rdi** [comment]
@coddec 
You need to build _Setup_ configuration, first 64bit then 32bit.

I'm trying to get _Appveyor_ builds working (this is my first experience with it). Though I cannot choose _Win32_ platform in settings. There is only _x86_ and _x64_ (which builds just fine btw).

So I'm running _BuildBinaries.bat_ directly. Here is _appveyor.yml_ config that I'm using currently:
```
version: 1.0.{build}
image: Visual Studio 2017
shallow_clone: true
build_script:
- cmd: >-
    cd ClassicShellSrc\ClassicShellSetup
    BuildBinaries.bat
```
This produces just binaries.
https://ci.appveyor.com/project/ge0rdi/classic-shell/build/1.0.3

I'm also trying to build full installer with _Appveyor_. But this requires _WiX_ (should be installed, but doesn't work for me right now) and HTML Help Workshop (this doesn't seem to be installed in build environment). Also I guess we'd need to replace _WinRar_ with _7zip_.

**Comment by coddec** [comment]
@XenHat  Ummm, I think it did, cause I deleted previous build.
@ge0rdi My first experience too :D
Yes, I guess we need to replace WinRar with 7zip

Here's my current appveyor.yml
`version: 4.3.1.{build}
pull_requests:
  do_not_increment_build_number: true
branches:
  only:
  - master
image: Visual Studio 2017
build:
  project: C:\projects\classic-shell\ClassicShellSrc\ClassicShell.sln
  verbosity: minimal`

**Comment by ge0rdi** [comment]
@coddec 
I've pushed few changes to [appveyor](https://github.com/coddec/Classic-Shell/compare/appveyor) branch.
It is now possible to build full installer from that branch using this config:
```
version: 4.4.{build}
pull_requests:
  do_not_increment_build_number: true
image: Visual Studio 2017
shallow_clone: true
install:
- cmd: set PATH=C:\Program Files (x86)\WiX Toolset v3.11\bin;C:\Program Files (x86)\HTML Help Workshop;%PATH%
build_script:
- cmd: >-
    cd ClassicShellSrc\ClassicShellSetup

    __MakeFinal.bat
test: off
artifacts:
- path: ClassicShellSrc\ClassicShellSetup\Final\*.exe
```
Look here:
https://ci.appveyor.com/project/ge0rdi/classic-shell/build/4.4.8/artifacts
(hope you can access it)

I will push that config to the branch, just want to check few things out.

Also not sure about the versioning. I'd suggest to continue from next minor version (4.4).
Next thing in my TODO list is to use _appveyor's_ version in actual build - version info of executables, name of installer, etc.

**Comment by coddec** [comment]
@ge0rdi Sure, you can push it anytime.
@ me if there's any pushing/permission issue.
Cheers!
